app/javascript/components/organisms/StatusHistoryPopover.js

```diff
@@ -0,0 +1,70 @@
+import React, { useState } from 'react'
+import classNames from 'classnames'
+import PropTypes from 'prop-types'
+
+import Icon from '~/components/atoms/Icon'
+import Popover from '~/components/molecules/Popover'
+import StatusItems from '~/components/molecules/lease_up_sidebar/StatusItems'
+import Loading from '../molecules/Loading'
+import apiService from '~/apiService'
+
+const MAX_UPDATES_TO_SHOW_DEFAULT = 4
+const StatusHistoryPopover = ({ applicationId }) => {
+  const [showingAllStatuses, setShowingAllStatuses] = useState(false)
+  const onShowHideStatusesToggled = () => setShowingAllStatuses(!showingAllStatuses)
+
+  const StatusIconButton = ({ onClick, ref }) => (
+    <button ref={ref} onClick={onClick} className='tiny icon margin-left--half'>
+      <Icon icon='list-unordered' size='medium' />
+    </button>
+  )
+  const [data, setData] = useState({ statusItems: [], loading: false })
```

Comment:
I think it's common to just call it `[state, setState]` when you're storing everything in one state object.

Only thing to be aware of is it's a little more fragile when stored as a big object, since you need to remember to set all state fields. If you ever just did setData({ loading: true }) it would wipe out your status items state.

Reply:
just figured in this case that the loading and the status items are pretty intertwined where we always want to update 1 with the other. I can split it out if that makes more sense

Reply:
That makes sense to me, fine to keep it as-is!